docs/usage/language-constraints-upgrading.md

```diff
@@ -0,0 +1,87 @@
+# Language Constraints and Upgrading
+
+## Package Releases have language constraints
+
+Many ecosystems support the concept that each release of packages has its own language "constraint".
+For example, a npm package may support Node.js 18 and 20 in its v1 releases and Node.js 20 and 22 from v2.0.0 onwards.
+
+In an ideal scenario:
+
+- Package files allow a project to declare its supported language constraints, and
+- Package registries allow packages to declare declare the supported language constraints per release
+
+## Restricting upgrades to compatible releases
+
+By default Renovate _does not_ apply language constraints to upgrades, which means it will propose "any" stable upgrade without consideration of language version.
+For certain ecosystems, changes to language constraints are usually done in major releases and documented in the release notes, so this behavior may not be problematic.
+For other ecosystems this may seem simply _wrong_.
+
+Renovate users can opt into strict compatibility filtering using `constraintsFiltering=strict`.
+You should understand the limitations of this first, also as an understanding of why it's not the default behavior.
+
+## Language constraint updating
+
+The first challenge is that Renovate may not yet support the ability to update your language constraints in an automated manner, and even when it does, users may not understand how many updates are depending on it.
+
+For example, consider a Node.js project which has its `engines` set to `"node": "^18.0.0 || ^20.0.0"`.
+
+Should Renovate skip Node.js v21 because it's non-LTS?
+When Node.js v22 comes out, should Renovate add it to your engines, or wait until it starts LTS?
+When Node.js v18 is EOL, should Renovate drop it from the engines list?
+
+It's hard for Renovate to know or guess what users want, because users have strong but also different opinions on the answer to the above questions.
+
+Additionally, even if Renovate guesses right or adds advanced capabilities to allow this to be configurable, then users might sit any of these "major" upgrades for months.
+If a project delays creating or merging the update to drop Node.js v18 from engines then it won't be able to upgrade to any new versions of library dependencies which themselves already dropped support.
+
+## Strict filtering limitations
+
+Consider again the Node.js project which has its `engines` set to `"node": "^18.0.0 || ^20.0.0"`.
+
+Now also consider a library which sets its `engines` to `"node": "^18.12.0 || ^20.9.0"` because it wants to only support "LTS releases" of Node.js.
```

Comment:
```suggestion
Now also consider a library which sets its `engines` field to `"node": "^18.12.0 || ^20.9.0"` because the library only supports "LTS releases" of Node.js.
```